Allow for event deletion

agentd00nut opened the pull request:
Multiple calendars can be added to the sourceCalendars global.

Events found in a users calendar not found in a feed during a sync can be removed, controlled by global flag.

Events no longer store feed ids in the description and use a custom "FID" metadata tag.

Event matching is based on this "FID" tag.

agentd00nut committed:
Multiple calendars and extended features.

Process multiple calendars feeds during each run.
Store feed event ids in the "FID" metadata tag instead of events description.
Allow for event deletion.  Deletes events that are found in current calendar if not found in feed.

agentd00nut committed:
Multiple calendars and extended features.

Process multiple calendars feeds during each run.
Store feed event ids in the "FID" metadata tag instead of events description.
Allow for event deletion. Deletes events that are found in current calendar if not found in feed.

agentd00nut committed:
Modify events when source event updates.

Modifies an event in the calendar if the source event with matching ID changes its title, description, startTime, endTime, or location

derekantrican commented on ICS-ICAL Sync/Code.gs:
```diff
@@ -20,13 +20,14 @@ var sourceCalendars={
 var addAlerts = true; //Whether to add the ics/ical alerts as notifications on the Google Calendar events
 var descriptionAsTitles = false; //Whether to use the ics/ical descriptions as titles (true) or to use the normal titles as titles (false)
-var defaultDuration = 60; //Default duration (in minutes) in case the event is missing an end specification in the ICS/ICAL file
+var defaultDuration = 30; //Default duration (in minutes) in case the event is missing an end specification in the ICS/ICAL file
```
What is the argument for making this change? I don't see a reason to change the default value of `defaultDuration`

agentd00nut replied:
No good reason.  Just forgot i changed it.

derekantrican commented on ICS-ICAL Sync/Code.gs:
```diff
@@ -4,20 +4,34 @@
+//     "targetCalendarName" is The name of the Google Calendar you want to add events to
+//     "sourceCalendarURL" is The ics/ical url that you want to get events from
+
+var sourceCalendars={
```
Let's do this change (support for multiple calendars) in a separate pull request. Log an issue for it

derekantrican commented on ICS-ICAL Sync/Code.gs:
```diff
@@ -4,20 +4,34 @@
 var addEventsToCalendar = true; //If you turn this to "false", you can check the log (View > Logs) to make sure your events are being read correctly before turning this on
+var modifyExistingEvents = true; // If you turn this to "false", any event in the feed that was modified after being added to the calendar will not update
+var removeEventsFromCalendar = true; //If you turn this to "true", any event in the calendar not found in the feed will be removed.  
+var createMissingCalendar=true;  // If the calendar name is not found we create it instead of throwing an error, only if the sourceUrl works.
```
I think we should just do this anyway. Don't have it be a user setting, just create the calendar if doesn't exist

derekantrican commented on ICS-ICAL Sync/Code.gs:
```diff
@@ -108,33 +146,49 @@ function main() {
-    
-    Logger.log("");
-  }
+//  for (var i = 0; i < events.length; i++){
```
Why were these lines (159-170) commented out? A change made during your testing?

agentd00nut replied:
Sorry yea, i kept having to scroll past all the output when debugging and forgot to flip it back.

derekantrican commented on ICS-ICAL Sync/Code.gs:
```diff
@@ -32,28 +46,50 @@ var email = ""; //OPTIONAL: If "emailWhenAdded" is set to true, you will need to
-function main() {  
+function main(){
+  
+  for( var calendar in sourceCalendars){
```
Again, let's do multi-calendar support in a different pull request

derekantrican commented on ICS-ICAL Sync/Code.gs:
```diff
@@ -32,28 +46,50 @@ var email = ""; //OPTIONAL: If "emailWhenAdded" is set to true, you will need to
-  if(targetCalendar == null)
-    throw "[ERROR] Could not find calendar of name \"" + targetCalendarName + "\"";
+  if(targetCalendar == null){
+    if(createMissingCalendar){
```
Like above comment - just do this. Don't need a user setting for it

agentd00nut committed:
Changes based on review.

Remove multi calendar support for a different pull request.
Return default interval time to 60 minutes
Remove missing calendar creation user flag.
Log detailed event info.

derekantrican commented:
@agentd00nut I want to revive this again and get it pulled through. I'll have some time this afternoon (around 4pm-6pm PST). Here's a Slack channel where we can discuss this better: https://join.slack.com/t/gasicssync/shared_invite/enQtMzc2MTk4MTkzOTEwLTViNGU4NjhkZmRhMjliMzY2NjY2ZWJjNDdkZGY3NTBiMTlmMTViMDZiZTE0NGVkMjFmMTU3NzkzZWQ2ZGY3NjM

agentd00nut committed:
Restoring Original timing variables

derekantrican commented:
Resolved #5
